File: numbers/src/main/java/org/dicio/numbers/parser/lexer/DurationToken.java
package org.dicio.numbers.parser.lexer;

import org.dicio.numbers.unit.Duration;

public class DurationToken extends Token {
    private final String durationCategory;
    private final Duration durationMultiplier;
    private final boolean restrictedAfterNumber;

    DurationToken(final String value,
                  final String spacesFollowing,
                  final String durationCategory,
                  final Duration durationMultiplier,
                  final boolean restrictedAfterNumber) {
        super(value, spacesFollowing);
        this.durationCategory = durationCategory;
        this.durationMultiplier = durationMultiplier;
        this.restrictedAfterNumber = restrictedAfterNumber;
    }

    @Override
    public boolean isDurationToken() {
        return true;
    }

    @Override
    public DurationToken asDurationToken() {
        return this;
    }

    public String getDurationCategory() {
        return durationCategory; // this is basically the duration multiplier, but in plain text
    }

    public Duration getDurationMultiplier() {
        return durationMultiplier;
    }

    public boolean isRestrictedAfterNumber() {
        return restrictedAfterNumber;
    }
}

File: numbers/src/main/java/org/dicio/numbers/parser/lexer/DurationToken.kt
package org.dicio.numbers.parser.lexer

import org.dicio.numbers.unit.Duration

class DurationToken internal constructor(
    value: String,
    spacesFollowing: String,
    positionInOriginalString: Int,
    // this is basically the duration multiplier, but in plain text
    val durationCategory: String,
    val durationMultiplier: Duration,
    val isRestrictedAfterNumber: Boolean
) : Token(value, spacesFollowing, positionInOriginalString) {
    override val asDurationToken = this
}
